api/search_test.go

```diff
@@ -0,0 +1,231 @@
+			"From={{.From}};" +
+			"Size={{.Size}};" +
+			"Types={{.Types}};" +
+			"Index={{.Index}};" +
+			"Queries={{.Queries}};" +
+			"SortBy={{.SortBy}};" +
+			"AggregationField={{.AggregationField}};" +
+			"Highlight={{.Highlight}};" +
+			"FilterOnLatest={{.FilterOnLatest}};" +
+			"FilterOnFirstLetter={{.FilterOnFirstLetter}};" +
+			"ReleasedAfter={{.ReleasedAfter}};" +
+			"ReleasedBefore={{.ReleasedBefore}};" +
+			"UriPrefix={{.UriPrefix}};" +
+			"Topic={{.Topic}};" +
+			"TopicWildcard={{.TopicWildcard}};" +
+			"Upcoming={{.Upcoming}};" +
+			"Published={{.Published}};" +
+			"Now={{.Now}}")
+		esMock := &ElasticSearcherMock{
+			MultiSearchFunc: func(index string, docType string, request []byte) ([]byte, error) {
+				return []byte(`{"dummy":"response"}`), nil
+			},
+		}
+
+		searchHandler := SearchHandlerFunc(esMock)
+
+		req := httptest.NewRequest(
+			"GET",
+			"http://localhost:8080/search?term=a"+
+				"&from=1"+
+				"&size=2"+
+				"&type=ta"+
+				"&type=tb"+
+				"&index=i"+
+				"&sort=date"+
+				"&query=qa"+
+				"&query=qb"+
+				"&aggField=_af"+
+				"&highlight=false"+
+				"&latest=true"+
+				"&withFirstLetter=f"+
+				"&releasedAfter=ra"+
+				"&releasedBefore=rb"+
+				"&uriPrefix=u"+
+				"&topic=t1"+
+				"&topic=t2"+
+				"&topicWildcard=tw1"+
+				"&topicWildcard=tw2"+
+				"&upcoming=true"+
+				"&published=true",
+			nil)
+		resp := httptest.NewRecorder()
+
+		searchHandler.ServeHTTP(resp, req)
+
+		So(resp.Code, ShouldEqual, http.StatusOK)
+		So(resp.Body.String(), ShouldResemble, `{"dummy":"response"}`)
+		So(esMock.MultiSearchCalls(), ShouldHaveLength, 1)
+		actualRequest := string(esMock.MultiSearchCalls()[0].Request)
+		So(actualRequest, ShouldContainSubstring, "Term=a")
+		So(actualRequest, ShouldContainSubstring, "From=1")
+		So(actualRequest, ShouldContainSubstring, "Size=2")
+		So(actualRequest, ShouldContainSubstring, "Types=[ta tb]")
+		So(actualRequest, ShouldContainSubstring, "Index=i")
+		So(actualRequest, ShouldContainSubstring, "SortBy=date")
+		So(actualRequest, ShouldContainSubstring, "Queries=[qa qb]")
+		So(actualRequest, ShouldContainSubstring, "AggregationField=_af")
+		So(actualRequest, ShouldContainSubstring, "Highlight=false")
+		So(actualRequest, ShouldContainSubstring, "FilterOnLatest=true")
+		So(actualRequest, ShouldContainSubstring, "FilterOnFirstLetter=f")
+		So(actualRequest, ShouldContainSubstring, "ReleasedAfter=ra")
+		So(actualRequest, ShouldContainSubstring, "ReleasedBefore=rb")
+		So(actualRequest, ShouldContainSubstring, "UriPrefix=u")
+		So(actualRequest, ShouldContainSubstring, "Topic=[t1 t2]")
+		So(actualRequest, ShouldContainSubstring, "TopicWildcard=[tw1 tw2]")
+		So(actualRequest, ShouldContainSubstring, "Upcoming=true")
+		So(actualRequest, ShouldContainSubstring, "Published=true")
+		So(actualRequest, ShouldContainSubstring, "Now=20")
+
+	})
+}
+
+func setupTestTemplates(rawtemplate string) {
+	temp, _ := template.New("search.tmpl").Parse(rawtemplate)
```

Comment:
I would assign this error and check its nil:
```
temp, err := template.New("search.tmpl").Parse(rawtemplate)
So(err, ShouldBeNil)
...
```
If the test template setup fails now it will fail the test instead of dying quietly in the background.